src/Counter.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import Button from './Button';
+
+export default function Counter() {
+  const initialState = { count: 0 };
+  const [state, setState] = useState(initialState);
+  const steps = [1, 2, 3, 4, 5];
+
+  function increaseCount(step) {
+    setState(({ count }) => ({
+      count: count + step,
+    }));
+  }
```

Comment:
```suggestion
  const { count } = state;
  
  function increaseCount(step) {
    setState({
      count: count + step,
    });
  }
```

표현 방식의 차이겠지만, 상위에서 구조분해할당을 한 뒤에 사용하는 방법도 가독성에 좋을 것 같군요

Reply:
이 부분은 조금 고민이 되네요. 제 생각은 이렇습니다. 

먼저 `increaseCount`부분은 `set`함수에 콜백을 주는 것이 이전 값을 활용해 다음값을 계산한다는 것이 더 명시적으로 느껴집니다. 만약 `state.count`값이 많이 재사용된다면 상위에서 미리 구조분해할당하는 것이 좋을 것 같은데 `increaseCount`부분 외에는 아래 부분 하나만 재사용됩니다.

```javascript
<Button
  value={`Click me! (${state.count})`}
  onClick={() => increaseCount(1)}
/>
```

그런데 이 버튼 부분은 리펙터링 책의 변수 인라인하기의 예시와 유사해보입니다. 아래는 그 예시입니다.
```javascript
// 이전
let basePrice = anOrder.basePrice;
return (basePrice > 1000);

// 이후
return anOrder.basePrice > 1000;
```
저자는 이를 `변수는 대체로 긍정적인 효과를 주지만, 그 이름이 원래 표현식과 다를 바 없을 때도 있다.` 라고 설명합니다. 

이러한 이유로 저는 그대로 두는 것이 더 나은 것 같은데 트레이너님의 의견은 어떠신가요?

Reply:
이 코멘트를 작성한 뒤에 [2번째 과제](https://github.com/CodeSoom/react-week2-assignment-2/pull/70#discussion_r629481747)를 수정하면서 

```javascript
const [state, setState] = useState({
  userInput: '',
  tasks: [],
});

const { userInput, tasks } = state;
```

제안해주신 방식으로 코드를 짜봤는데 이 경우에는 반복이 많으니 이런식으로 하는게 확실히 좋네요. 
코드의 복잡도가 늘어날수록 제안해주신 쪽이 더 낫다고도 느껴지네요.

Reply:
말씀하신대로 관리하는 상태가 적을 경우에는 작성하신 것처럼 콜백을 사용한 방식이 의도가 더 드러날 수 있을 것 같습니다 👍🏼 

코드의 의도와 가독성 등 여러 관점에서 고민해보시고 적용해주셨군요 :)

저도 유종님 덕분에 코드의 복잡도에 따라 어떤 방식이 좋은지 이번 기회에 고민해볼 수 있었네요. 🙏🏻